src/devices/cpu/mipsx/mipsxdasm.cpp

```diff
@@ -0,0 +1,637 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+#include "emu.h"
+#include "mipsxdasm.h"
+
+u32 mipsx_disassembler::opcode_alignment() const
+{
+	return 4;
+}
+
+int mipsx_disassembler::get_ty(u32 opcode)
+{
+	return (opcode & 0xc0000000) >> 30;
```

Comment:
I can do, but as I've said before, I tend to find util::BIT() to be a form of obfuscation myself and tend to avoid it.  It makes the code harder to read.